File: matiasmasso/Mat.NET/DTO/Extensions/EmptyStringDataAnnotationsModelMetadataProvider.cs
using System;
using System.Collections.Generic;
using System.Text;
using System.Web.Mvc;

namespace DTO.Extensions
{
    public class EmptyStringDataAnnotationsModelMetadataProvider:System.Web.Mvc.DataAnnotationsModelMetadataProvider
    {    
        protected  override ModelMetadata CreateMetadata(IEnumerable<Attribute> attributes, Type containerType, Func<object> modelAccessor, Type modelType, string propertyName)
        {
            var modelMetadata = base.CreateMetadata(attributes, containerType, modelAccessor, modelType, propertyName);
            modelMetadata.ConvertEmptyStringToNull = false;
            return modelMetadata;
        }
    }
}

File: matiasmasso/Mat.NET/Website/Helpers/EmptyStringDataAnnotationsModelMetadataProvider.vb
Public Class EmptyStringDataAnnotationsModelMetadataProvider
    Inherits System.Web.Mvc.DataAnnotationsModelMetadataProvider

    Protected Overrides Function CreateMetadata(ByVal attributes As IEnumerable(Of Attribute), ByVal containerType As Type, ByVal modelAccessor As Func(Of Object), ByVal modelType As Type, ByVal propertyName As String) As ModelMetadata
        Dim modelMetadata = MyBase.CreateMetadata(attributes, containerType, modelAccessor, modelType, propertyName)
        modelMetadata.ConvertEmptyStringToNull = False
        Return modelMetadata
    End Function
End Class
